packages/x-tree-view/src/TreeItem2DragAndDropOverlay/TreeItem2DragAndDropOverlay.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { alpha, styled } from '@mui/material/styles';
+import { shouldForwardProp } from '@mui/system';
+import { TreeItem2DragAndDropOverlayProps } from './TreeItem2DragAndDropOverlay.types';
+import { TreeViewItemsReorderingAction } from '../models';
+
+const TreeItem2DragAndDropOverlayRoot = styled('div', {
+  name: 'MuiTreeItem2DragAndDropOverlay',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+  shouldForwardProp: (prop) => shouldForwardProp(prop) && prop !== 'action',
+})<{ action?: TreeViewItemsReorderingAction | null }>(({ theme }) => ({
+  position: 'absolute',
+  left: 0,
+  display: 'flex',
+  top: 0,
+  bottom: 0,
+  right: 0,
+  pointerEvents: 'none',
+  variants: [
+    {
+      props: { action: 'make-child' },
+      style: {
+        marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
+        borderRadius: theme.shape.borderRadius,
+        backgroundColor: alpha((theme.vars || theme).palette.primary.dark, 0.15),
+      },
+    },
+    {
+      props: { action: 'reorder-above' },
+      style: {
+        marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
+        borderTop: `1px solid ${alpha((theme.vars || theme).palette.grey[900], 0.6)}`,
+        ...(theme.palette.mode === 'dark' && {
+          borderTop: `1px solid ${alpha((theme.vars || theme).palette.grey[100], 0.6)}`,
+        }),
+      },
+    },
+    {
+      props: { action: 'reorder-below' },
+      style: {
+        marginLeft: 'calc(var(--TreeView-indentMultiplier) * var(--TreeView-itemDepth))',
+        borderBottom: `1px solid ${alpha((theme.vars || theme).palette.grey[900], 0.6)}`,
+        ...(theme.palette.mode === 'dark' && {
+          borderBottom: `1px solid ${alpha((theme.vars || theme).palette.grey[100], 0.6)}`,
+        }),
+      },
+    },
+    {
+      props: { action: 'move-to-parent' },
+      style: {
+        marginLeft:
+          'calc(var(--TreeView-indentMultiplier) * calc(var(--TreeView-itemDepth) - 1))' as any,
+        borderBottom: `1px solid ${alpha((theme.vars || theme).palette.grey[900], 0.6)}`,
+        ...(theme.palette.mode === 'dark' && {
+          borderBottom: `1px solid ${alpha((theme.vars || theme).palette.grey[900], 0.6)}`,
+        }),
+      },
+    },
+  ],
+}));
+
+function TreeItem2DragAndDropOverlay(props: TreeItem2DragAndDropOverlayProps) {
```

Comment:
This is on the community package because it allows to have a single Tree Item component shared across all plans.
For now I did not try to add any shiny abstraction to have it in the pro package, I think having a dummy component here is the best trade-off

If we want to go a little further, we can see how to inject the style from the pro plan since it's almost all the this component currently have, but I don't think it's worth the effort for now